Automattic/jetpack
# WP Super Cache: make preload faster and more robust

The preload function in WP Super Cache allows a website owner to prime the cache on their site by creating cache files for all the taxonomy archives (first page), pages, and posts. It does this by fetching each page separately.

* Instead of getting 100 pages at a time, it will request 10 instead. This is because some sites are very slow or have pages that are slow to generate. If you're using FastCGI then set_time_limit() doesn't reset the read timeout.
* It will preload all taxonomies first now. This avoids a problem where you have lots of taxonomies and not enough posts to run the preload loop enough times to preload everything. The preload loop used to run on the number of posts on the site.
* I removed many pauses as it's only 10 pages per preload loop now, fetched sequentially, with a 3 second pause in between.
* The preload notification JS only requests every 3 seconds now as there's a 3 second wait between jobs, it avoids requesting a notification with the same filename over and over again.

https://wordpress.org/support/topic/preload-mode-skips-caching-pages-or-deletes-already-cached-pages/#post-16706554

## Testing instructions:
You can copy wp-cache.php from this branch into a live site. Also copy inc/preload-notifcation.js but that was a very minor change.
Run a preload and note that taxonomies are cached first.

### Other information:



## Does this pull request change what data or activity we track or use?
No
